numpy/ma/extras.py

```diff
@@ -708,34 +708,39 @@ def _median(a, axis=None, out=None, overwrite_input=False):
             asorted = a
     else:
         asorted = sort(a, axis=axis)
+
     if axis is None:
         axis = 0
     elif axis < 0:
         axis += a.ndim
+    if axis < 0 or axis >= a.ndim:
+        raise ValueError("Axis value out of bounds")
 
     if asorted.ndim == 1:
         idx, odd = divmod(count(asorted), 2)
-        return asorted[idx - (not odd) : idx + 1].mean()
+        return asorted[idx + odd - 1 : idx + 1].mean()
```

Comment:
It is a bit strange that below (https://github.com/numpy/numpy/pull/8016/files#diff-6fd4523c4f6741d8d9e9a20d14561729R745) for inexact dtype the mean is done by summing and then dividing, and here this is ignored.  It is out of scope of this PR, though.